### PR TITLE
feat: add resource show command

### LINKED ISSUE #1261: Feat: kusion resource command
### What would you like to be added?

Command `kusion resource` to interact with resources :
- [ ] ~Add new command `kusion resource list` to show all resources of a specific project in a workspace.~
      - `--workspace`: optional, default is current workspace
      - `--project`: required
      - `--backend`: optional, default is current backend
- [ ] Add new command `kusion resource show {id}` to show the latest state of the specific resource by the id.
      - `--workspace`: optional, default is current workspace
      - `--project`: required
      - `--backend`: optional, default is current backend
### Why is this needed?
- It serves to check the status and get resource information by ID, by project, by workspace, or by all resources.
- To get the information about resources of a project that are applied in a workspace, from that we can get the `kusion-id` and the terraform import id of these resources and use them for another project in the same workspace.

### PR DESCRIPTION
**Pull request by hoangndst**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:
Show details of a resource of the current or specified stack.

#### Which issue(s) this PR fixes:

Fixes #1261

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by hoangndst** [comment]
cc @Yangyang96 @ffforest @liu-hm19

File: pkg/cmd/resource/show.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	for _, res := range rel.Spec.Resources {
+		resourceMap[res.ResourceKey()] = res
+	}
+
+	stateMap := make(map[string]apiv1.Resource)
+	for _, res := range rel.State.Resources {
+		stateMap[res.ResourceKey()] = res
+	}
+
+	if _, ok := resourceMap[*o.ID]; !ok {
```
Comment (liu-hm19) [inline pkg/cmd/resource/show.go:217]: Is it necessary to consider a special corner case here, where a specified resource in the `Release` file is not in the `Spec` but still exists in the `State`? For example, the last `kusion destroy` failed. And if we consider this situation, we might not be able to return an error directly here. cc @Yangyang96

**Review by liu-hm19 — APPROVED** [review]

File: pkg/cmd/resource/show.go (hunk 1 above)
Comment (Yangyang96) [reply pkg/cmd/resource/show.go:217]: > Is it necessary to consider a special corner case here, where a specified resource in the `Release` file is not in the `Spec` but still exists in the `State`? For example, the last `kusion destroy` failed. And if we consider this situation, we might not be able to return an error directly here. cc @Yangyang96

After discussing with @liu-hm19, we have decided to leave the code as it is and set aside this particular corner case. The example above did not sufficiently explain this situation. This case arises when a user attempts to remove a resource using the `kusion apply` operation but it fails, while other commands such as `kusion resource graph` and `kusion destroy` do not handle it differently. This limitation is imposed by the structure of the command-line itself, and `kusion` will not be able to address all of these corner cases. As a result, we have decided to retain your implementation as it accurately reflects the initial user intentions. @hoangndst It's your decision to make to leave it here or change your implementation.

**Review by Yangyang96 — APPROVED** [review]

File: pkg/cmd/resource/show.go (hunk 1 above)
Comment (hoangndst) [reply pkg/cmd/resource/show.go:217]: For the original purpose of `kusion resource show`, in my opinion, we should keep this approach. 
Thank you for taking the time to provide your comments. 
cc @liu-hm19 @Yangyang96

Comment (liu-hm19) [reply pkg/cmd/resource/show.go:217]: OK 👍 We agree with you. 

And during the discussion about the corner case, we've found a bug concerning the resource elimination in the `State` of the `Release` file when executing the `kusion destroy` command 😂 We will open an issue to track and fix this bug in the future.

**Pull request merged**